src/c-writer.cc

```diff
@@ -1085,40 +1085,12 @@ void CWriter::WriteTags() {
   for (const Tag* tag : module_->tags) {
     bool is_import = tag_index < module_->num_tag_imports;
     if (!is_import) {
-      Write("static u32 ",
+      Write("static const char ",
```

Comment:
Perhaps add a comment here?  Something like "tags are identified and compared by their address alone, so the type used here and the data stored at the address are irrelevant. "?

Reply:
I wonder if we should also at `typedef Tag char` and use that so avoid readers being confused by the use of   char here and the use of `char*` below (when I see that I wonder what string data is stored, and if its null terminated, etc).

Then the comment above could be attached to the typedef.

Reply:
Done -- I made the actual variable a wasm_tag_placeholder_t (which is private to the .c file) and the public type is a wasm_rt_tag_t (typedef'd to void* in wasm-rt.h). Hope that's compatible with what you had in mind, otherwise happy to amend.